File: Software Technologies/04.Java/03.Java_Spring_Calculator/src/main/java/com/softuni/controller/HomeController.java
package com.softuni.controller;

import com.softuni.entity.Calculator;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;

@Controller
public class HomeController {
	@GetMapping("/")
	public String index(Model model) {
		model.addAttribute("operator", "+");
		model.addAttribute("view","views/calculatorForm");
		return "base-layout";
	}

	@PostMapping("/")
	public String calculate(Model model, Calculator calculator) {
		model.addAttribute("operator", calculator.getOperator());
		model.addAttribute("rightOperand", calculator.getRightOperand());
		model.addAttribute("leftOperand", calculator.getLeftOperand());
		model.addAttribute("result", calculator.calculateResult());
		model.addAttribute("view","views/calculatorForm");
		return "base-layout";
	}
}

File: Software Technologies/05.C#/03.Calculator_C#/Calculator-CSharp/Controllers/HomeController.cs
using System;
using System.Web.Mvc;
using Calculator_CSharp.Models;

namespace Calculator_CSharp.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index(Models.Calculator calculator)
        {
            return View(calculator);
        }

        [HttpPost]
        public ActionResult Calculate(Models.Calculator calculator)
        {
            calculator.Result = CalculateResult(calculator);
            return RedirectToAction("Index", calculator);
        }

        private decimal CalculateResult(Calculator calculator)
        {
            var result = 0m;

            switch (calculator.Operator)
            {
                case "+": result = calculator.LeftOperand + calculator.RightOperand; break;
                case "-": result = calculator.LeftOperand - calculator.RightOperand; break;
                case "*": result = calculator.LeftOperand * calculator.RightOperand; break;
                case "/": result = calculator.LeftOperand / calculator.RightOperand; break;

                default:
                    break;
            }
            return result;
        }
    }
}